equinix/resource_network_device.go

```diff
@@ -635,6 +743,70 @@ func createNetworkDeviceUserKeySchema() map[string]*schema.Schema {
 	}
 }
 
+func createClusterNodeDetailRequestSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		neDeviceClusterSchemaNames["LicenseFileId"]: {
+			Type:        schema.TypeString,
+			Optional:    true,
+			ForceNew:    true,
+			Sensitive:   true,
+			Description: neDeviceClusterDescriptions["LicenseFileId"],
+		},
+		neDeviceClusterSchemaNames["LicenseToken"]: {
+			Type:        schema.TypeString,
+			Optional:    true,
+			ForceNew:    true,
+			Sensitive:   true,
+			Description: neDeviceClusterDescriptions["LicenseToken"],
+		},
+		neDeviceClusterSchemaNames["VendorConfiguration"]: {
+			Type:     schema.TypeMap,
+			Optional: true,
+			ForceNew: true,
+			Elem: &schema.Schema{
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			Description: neDeviceClusterDescriptions["VendorConfiguration"],
+		},
+	}
+}
+
+func createNetworkDeviceClusterNodeSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		neDeviceClusterSchemaNames["UUID"]: {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: neDeviceClusterDescriptions["UUID"],
+		},
+		neDeviceClusterSchemaNames["Name"]: {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: neDeviceClusterDescriptions["Name"],
+		},
+		neDeviceClusterSchemaNames["Node"]: {
+			Type:        schema.TypeInt,
+			Computed:    true,
+			Description: neDeviceClusterDescriptions["Node"],
+		},
+		neDeviceClusterSchemaNames["AdminPassword"]: {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Sensitive:   true,
+			Description: neDeviceClusterDescriptions["AdminPassword"],
+		},
+		neDeviceClusterSchemaNames["VendorConfiguration"]: {
+			Type:     schema.TypeMap,
+			Computed: true,
+			Elem: &schema.Schema{
```

Comment:
https://developer.equinix.com/docs/launch-cluster-device
> {vendorConfig{hostName, activationKey, controllerFqdn, rootPassword, adminPassword, controller1}

It seems there is a predefined list of valid parameters. It would be better to use a TypeSet so that users know what values they can include

Reply:
This is an existing field under device schema as well and for now we won't change it because it's already consumed by the customer.

Reply:
I think the change to the user is something that can be introduced in a semantic version "minor" release. 

While this is technically a breaking change, the user fix is simple (remove the `=` sign). 

@rling-equinix points out that the access pattern will also change, the map would be accessed with an index: `vendor_config.field` becomes `vendor_config[0].field` or `vendor_config.0.field`. 

The user benefit is that the structure will be validated rather than be an anonymous list of strings. The contents of the fields in the structure can also be validated and strictly typed. We can also mark the sensitive vendor configs as such.

Reply:
https://www.terraform.io/plugin/sdkv2/best-practices/versioning#example-minor-number-increments

> Increasing the minor number suggests that new features have been added but that existing functionality remains broadly compatible.

However:
> Example Major Number Increments: ...
> * Changing attribute type where the new type is functionally incompatible (including but not limited to changing TypeSet to TypeList and TypeList to TypeSet)

Perhaps this change should be deferred on the EXISTING parameter. 

For this PR, we could introduce this as a list of schema map.  While the two identically named parameters would have a slightly different interface, we'd get the benefits described above.   What are your thoughts, @rling-equinix @ocobleseqx?

Reply:
We can definitely change the schema of vendorConfig for cluster, however, for top level vendorConfig under device shema, let's defer that changes.